src/components/helpers/wrapWithApiConnect.js

```diff
@@ -1,81 +1,62 @@
-import { Component, createElement } from 'react'
-
-import invariant from 'invariant'
 import hoistStatics from 'hoist-non-react-statics'
-import { forEach } from 'lodash'
+import React, { useEffect, useRef } from 'react'
 
 import { promisePropsEqual } from '../../utils'
-import getDisplayName from './getDisplayName'
 import { ELEMENT_ID_PROP_NAME } from './constants'
+import getDisplayName from './getDisplayName'
 
 const wrapWithApiConnect = ({
   finalMapPropsToPromiseProps,
-  withRef,
 }) => WrappedComponent => {
-  class ApiConnect extends Component {
-    componentWillMount() {
-      this.loadEntities(this.props)
-    }
+  function ApiConnect({ [ELEMENT_ID_PROP_NAME]: elementId, innerRef, ...rest }) {
+    const promisePropsRef = useRef({})
 
-    componentWillUpdate(nextProps) {
-      this.loadEntities(nextProps, this.props)
-    }
+    const promiseProps = finalMapPropsToPromiseProps(rest)
+    const prevPromiseProps = promisePropsRef.current
 
-    render() {
-      const { [ELEMENT_ID_PROP_NAME]: elementId, ...rest } = this.props
-      return createElement(WrappedComponent, {
-        ...rest,
-        ...(withRef ? { ref: 'wrappedInstance' } : {}),
-      })
-    }
+    const fetchProps = Object.keys(promiseProps).reduce((props, propName) => {
+      const promiseProp = promiseProps[propName]
 
-    loadEntities(props, prevProps) {
-      const promiseProps = finalMapPropsToPromiseProps(props)
-      const prevPromiseProps = prevProps
-        ? finalMapPropsToPromiseProps(prevProps)
-        : {}
+      if (promiseProp.method !== 'fetch') {
+        return props
+      }
 
-      forEach(promiseProps, (promiseProp, propName) => {
-        const { method } = promiseProp
-
-        if (method !== 'fetch') {
-          return
-        }
+      return {
+        ...props,
+        [propName]: rest[propName]
+      }
+    }, {})
 
+    useEffect(() => {
+      Object.keys(fetchProps).forEach((propName) => {
+        const fetchProp = fetchProps[propName]
+        const promiseProp = promiseProps[propName]
         // always refresh on any change of the query or if the refresh token is set
         // and changed since the last render
         const promisePropUpdated =
           !prevPromiseProps[propName] ||
-          !promisePropsEqual(promiseProp, prevPromiseProps[propName])
+          !promisePropsEqual(promiseProps[propName], prevPromiseProps[propName])
 
-        const { fetchOnMount } = promiseProps[propName]
```

Comment:
No more support for `fetchOnMount` required? We should document this as a breaking change then.

Reply:
Ah I see you mentioned it in the PR as well, so 👍

Reply:
Yup. It wasn't documented anyway and we also never used it. However, I think moving forward it is good if we don't think in lifecycles so much. As far as I can tell the whole react dev team now actively discourages this and hooks also evolved in that direction.